src/js/common/schemaform/definitions.js

```diff
@@ -36,3 +37,60 @@ export const ssn = {
   type: 'string',
   pattern: '^([0-9]{3}-[0-9]{2}-[0-9]{4}|[0-9]{9})$'
 };
+
+export const date = {
+  pattern: '^(\\d{4}|XXXX)-(0[1-9]|1[0-2]|XX)-(0[1-9]|[1-2][0-9]|3[0-1]|XX)$',
+  type: 'string'
+};
+
+export const dateRange = {
+  type: 'object',
+  properties: {
+    from: date,
+    to: date
+  }
+};
+
+const countryValues = countries.map(object => object.value);
```

Comment:
This address code can be ignored, it'll be replace when we do the real Address field.